src/Component/Metadata/Resource/ResourceMetadataCollection.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Metadata\Resource;
+
+use Sylius\Component\Resource\Metadata\Operation;
+use Sylius\Component\Resource\Metadata\Resource as ResourceMetadata;
+
+final class ResourceMetadataCollection extends \ArrayObject
+{
+    public function getOperation(string $resourceAlias, string $name): ?Operation
+    {
+        /** @var ResourceMetadata $current */
+        foreach ($this->getIterator() as $current) {
+            if ($current->getAlias() !== $resourceAlias) {
```

Comment:
Why don't we treat alias as a key in the array?

Reply:
Yes, that's a good idea

Reply:
Now, the operation name equals the route name when it's an http operation. 
That's not easy cause the operation name is a concatenation and it's build later.
That's hard to build it earlier when we build the Operations via the nested operations.

```php
#[Resource(
    alias: 'app.dummy',
    section: 'admin',
    operations: [
        new Index(),
        new Create(),
    ],
)]
```

```php
final class Resource
{
    private ?Operations $operations;

    public function __construct(
        private string $alias,
        private ?string $section = null,
        ?array $operations = null,
    ) {
        $this->operations = null === $operations ? null : new Operations($operations);
    }
```

```php
final class Operations implements \IteratorAggregate, \Countable
{
    private array $operations = [];

    public function __construct(array $operations = [])
    {
        foreach ($operations as $operationName => $operation) {
            $this->operations[] = [$operationName, $operation];
        }
    }

    // [...]
}
```